pennylane/tape/qscript.py

```diff
@@ -373,6 +373,61 @@ def op_wires(self) -> Wires:
 
     ##### Update METHODS ###############
 
+    def update(self, **kwargs):
+        r"""update(operations, measurements, shots, trainable_params)
+
+        Update attributes on the tape.
+
+        Keyword Args:
+            operations (Iterable[Operator]): An iterable of the operations to replace on the tape.
+            measurements (Iterable[MeasurementProcess]): An iterable of all the measurements to
+                replace on the tape.
+            shots (None, int, Sequence[int], ~.Shots): Number and/or batches of shots to replace
+                on the tape.
+            trainable_params (None, Sequence[int]): the indices for which parameters are trainable.
+
+        Returns: A new tape instance, initialized with any kwargs passed to update.
+            Anything not set in update will be the same as the initial tape.
+
+        Anything not set by update will retain the values on the original tape.
+
+        **Example**
+
+        .. code-block:: python
+
+            tape = qml.tape.QuantumScript(
+                ops= [qml.X(0), qml.Y(1)],
+                measurements=[qml.expval(qml.Z(0)],
+                shots=2000)
+
+            new_tape = tape.update(measurements=[qml.expval(qml.X(1))])
```

Comment:
How about making a `copy` method with an `update` keyword argument instead?

Reply:
Ooooh, I like it. I'll shuffle things around.